### PR TITLE
refreshVersionsMigrate use dependency constant names when available

### PR DESCRIPTION
**Pull request by jmfayard**
## 🚀 Description
One part we left out in `refreshVersionsMigrate` was to use the replace the maven coordinates with the matching dependency notation, if available.

Turns out it's possible to do it.

## 📄 Motivation and Context
Better migration

## 🧪 How Has This Been Tested?

https://github.com/jmfayard/ktor-samples/commit/e101b1cc00e31b4f271358ad199a15e18518655a

https://github.com/jmfayard/Anki-Android/commit/902edb9d203fe9c2e85572133b890a133c8d713e

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have read the [guidelines for the development process](https://jmfayard.github.io/refreshVersions/contributing/submitting-prs/dev-process/)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jmfayard** [comment]
The check failed is because `markDependencyNotationsUsage() ` doesn't make sense if we use all dependencies via `getArtifactNameToConstantMapping()`

**Comment by LouisCAD** [comment]
I knew it was possible and I told you 🙃

**Comment by LouisCAD** [comment]
Let's find a way to disable those checks while we are getting the mapping, I already have an idea on how we can do it.

**Comment by jmfayard** [comment]
Done, ready to review.
The diff seems big but it's mostly that `RefreshVersionsMigrateTask` and its tests had to be moved from core to dependencies

File: plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/DependencyGroup.kt [hunk 1]
```diff
@@ -27,6 +27,7 @@ open class DependencyGroup(
         val ALL_RULES: List<ArtifactVersionKeyRule>
             get() = ALL.mapNotNull { it.rule }
 
+        var disableBomCheck: Boolean = false
```
Comment (LouisCAD) [inline plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/DependencyGroup.kt:30]: Shouldn't this property be `internal`, or `@InternalRefreshVersionsApi`?

Comment (jmfayard) [reply plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/DependencyGroup.kt:33]: done, ,thanks

**Pull request merged**